plugin/pkg/admission/serviceplan/defaultserviceplan/admission.go

```diff
@@ -0,0 +1,123 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package defaultserviceplan
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apiserver/pkg/admission"
+
+	informers "github.com/kubernetes-incubator/service-catalog/pkg/client/informers_generated/internalversion"
+	internalversion "github.com/kubernetes-incubator/service-catalog/pkg/client/listers_generated/servicecatalog/internalversion"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+	scadmission "github.com/kubernetes-incubator/service-catalog/pkg/apiserver/admission"
+)
+
+const (
+	// PluginName is name of admission plug-in
+	PluginName = "DefaultServicePlan"
+)
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(io.Reader) (admission.Interface, error) {
+		return NewDefaultServicePlan()
+	})
+}
+
+// exists is an implementation of admission.Interface.
+// It checks to see if Service Instance is being created without
+// a Service Plan if there is only one Service Plan for the
+// specified Service and defaults to that value.
+// that the cluster actually has support for it.
+type defaultServicePlan struct {
+	*admission.Handler
+	scLister internalversion.ServiceClassLister
+}
+
+var _ = scadmission.WantsInternalServiceCatalogInformerFactory(&defaultServicePlan{})
+
+func (d *defaultServicePlan) Admit(a admission.Attributes) error {
+	// we need to wait for our caches to warm
+	if !d.WaitForReady() {
+		return admission.NewForbidden(a, fmt.Errorf("not yet ready to handle request"))
+	}
+
+	// We only care about service Instances
+	if a.GetResource().Group != servicecatalog.GroupName || a.GetResource().GroupResource() != servicecatalog.Resource("instances") {
+		return nil
+	}
+	instance, ok := a.GetObject().(*servicecatalog.Instance)
+	if !ok {
+		return apierrors.NewBadRequest("Resource was marked with kind Instance but was unable to be converted")
+	}
+	// If the plan is specified, let it through and have the controller
+	// deal with finding the right plan, etc.
+	if len(instance.Spec.PlanName) > 0 {
+		return nil
+	}
+
+	sc, err := d.scLister.Get(instance.Spec.ServiceClassName)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return admission.NewForbidden(a, err)
+		}
+		msg := fmt.Sprintf("ServiceClass %q does not exist, can not figure out the default Service Plan.", instance.Spec.ServiceClassName)
+		glog.V(4).Info(msg)
+		return admission.NewForbidden(a, errors.New(msg))
+	}
+	if len(sc.Plans) > 1 {
+		msg := fmt.Sprintf("ServiceClass %q has more than one plan, PlanName must be specified", instance.Spec.ServiceClassName)
+		glog.V(4).Info(msg)
+		return admission.NewForbidden(a, errors.New(msg))
+	}
+
+	p := sc.Plans[0]
+	glog.V(4).Infof("Using default plan %s for Service Class %s for instance %s",
```

Comment:
can you convert the %s here, and elsewhere, to %q's so things are quoted and easier to read?